dspace-api/src/test/java/org/dspace/app/util/GoogleBitstreamComparatorTest.java

```diff
@@ -0,0 +1,361 @@
+        assertEquals("PS should be first", "bitstream3", toSort.get(0).getName());
+        assertEquals("RTF second", "bitstream2", toSort.get(1).getName());
+        assertEquals("Word third", "bitstream1", toSort.get(2).getName());
+    }
+
+    /**
+     * Test to check for no crash when nothing is configured, just checks for size
+     */
+    @Test
+    public void testNoConfig() throws Exception{
+        settings.remove("citation.prioritized_types");
+        when(bitstreamFormat1.getMIMEType()).thenReturn("text/richtext");
+        when(bitstreamFormat2.getMIMEType()).thenReturn("application/msword");
+        when(bitstreamFormat3.getMIMEType()).thenReturn("application/postscript");
+        when(bitstream1.getSize()).thenReturn(new Long(100));
+        when(bitstream2.getSize()).thenReturn(new Long(200));
+        when(bitstream3.getSize()).thenReturn(new Long(300));
+
+        List<Bitstream> toSort = bundle.getBitstreams();
+        Collections.sort(toSort, new GoogleBitstreamComparator(context, settings));
+        assertEquals("bitstream3", toSort.get(0).getName());
+        assertEquals("bitstream2", toSort.get(1).getName());
+        assertEquals("bitstream1", toSort.get(2).getName());
+    }
+
+    /**
+     * Test to see what happens when you choose a short description that's not defined in bitstream-formats.xml
+     */
+    @Test
+    public void testUndefinedShortDescription() {
+        settings.put("citation.prioritized_types", "Postscript, RTF, Undefined Type, Microsoft Word, Adobe PDF");
+        when(bitstreamFormat1.getMIMEType()).thenReturn("text/richtext");
+        when(bitstreamFormat2.getMIMEType()).thenReturn("application/msword");
+        when(bitstreamFormat3.getMIMEType()).thenReturn("application/postscript");
+        when(bitstream1.getSize()).thenReturn(new Long(100));
+        when(bitstream2.getSize()).thenReturn(new Long(200));
+        when(bitstream3.getSize()).thenReturn(new Long(300));
+
+        List<Bitstream> toSort = bundle.getBitstreams();
+        Collections.sort(toSort, new GoogleBitstreamComparator(context, settings));
+        assertEquals("bitstream3", toSort.get(0).getName());
+        assertEquals("bitstream1", toSort.get(1).getName());
+        assertEquals("bitstream2", toSort.get(2).getName());
+    }
+
+    /**
+     * Test adding a new format in the priority list
+     */
+    @Test
+    public void testAddingNewFormat() {
+        settings.put("citation.prioritized_types", "WAV, Adobe PDF, Microsoft Word, RTF, Postscript");
+        when(bitstreamFormat1.getMIMEType()).thenReturn("text/richtext");
+        when(bitstreamFormat2.getMIMEType()).thenReturn("application/msword");
+        when(bitstreamFormat3.getMIMEType()).thenReturn("audio/x-wav");
+        when(bitstream1.getSize()).thenReturn(new Long(100));
+        when(bitstream2.getSize()).thenReturn(new Long(200));
+        when(bitstream3.getSize()).thenReturn(new Long(300));
```

Comment:
I'll inverse the sizes so that there is a double-check that type priorities overrule sizes.